src/__tests__/WeekView.test.js

```diff
@@ -1,10 +1,454 @@
+    const baseEventsOverlap = [
+      {
+        id: 9,
+        startDate: new Date(2020, 7, 24, 17),
+        endDate: new Date(2020, 7, 24, 18),
+      },
+      {
+        id: 43,
+        startDate: new Date(2020, 7, 24, 17),
+        endDate: new Date(2020, 7, 24, 18),
+      },
+      {
+        id: 11,
+        startDate: new Date(2020, 7, 25, 12, 15),
+        endDate: new Date(2020, 7, 25, 14, 30),
+      },
+      {
+        id: 2,
+        startDate: new Date(2020, 7, 25, 13, 0),
+        endDate: new Date(2020, 7, 25, 13, 45),
+      },
+      {
+        id: 3,
+        startDate: new Date(2020, 7, 24, 17),
+        endDate: new Date(2020, 7, 24, 18),
+      },
+      {
+        id: 1,
+        startDate: new Date(2020, 7, 26, 13),
+        endDate: new Date(2020, 7, 26, 20),
+      },
+    ].map((evt) => ({ ...evt, description: `event ${evt.id}` }));
+
+    const baseProps = {
+      numberOfDays: 3,
+      selectedDate: new Date(2020, 7, 24),
+      hoursInDisplay: 12,
+      startHour: 12,
+    };
+
+    it('shows all events', () => runTestShowsAll(baseEventsOverlap, baseProps));
+
+    it('shows each event with its color', () => {
+      const colors = [
+        'purple',
+        'red',
+        'pink',
+        'yellow',
+        'lightblue',
+        'lightblue',
+      ];
+      runTestShowColor(baseEventsOverlap, baseProps, colors);
+    });
+  });
+});
+
+describe('User interactions', () => {
+  describe('Pressing events', () => {
+    const targetId = 34;
+    const events = [
+      {
+        id: targetId,
+        startDate: new Date(2020, 7, 24, 12),
+        endDate: new Date(2020, 7, 24, 18),
+        description: 'target event',
+      },
+      {
+        id: 91,
+        startDate: new Date(2020, 7, 23, 20),
+        endDate: new Date(2020, 7, 23, 21, 15),
+        description: 'other',
+      },
+    ];
+
+    const mockRenderAndFire = () => {
+      const mockOnEventPress = jest.fn();
+
+      render(
+        <WeekView
+          events={events}
+          numberOfDays={5}
+          selectedDate={new Date(2020, 7, 22)}
+          hoursInDisplay={24}
+          startHour={0}
+          onEventPress={mockOnEventPress}
+        />,
+      );
+
+      /**
+       * NOTE: Ideally, the press would be grabbed without knowledge of gesture-handler,
+       * for example: `fireEvent.press(getByText('target event'))`,
+       * but RNGH does not provide an appropriate API for that
+       */
+      fireGestureHandler(getByGestureTestId(`pressGesture-${targetId}`), []);
+
+      return mockOnEventPress;
+    };
+
+    it('calls callback exactly once', () => {
+      const mockOnEventPress = mockRenderAndFire();
+      expect(mockOnEventPress).toHaveBeenCalledOnce();
+    });
+
+    it('calls callback with event as first argument', () => {
+      const mockOnEventPress = mockRenderAndFire();
+      const [eventArg] = mockOnEventPress.mock.calls[0];
+      expect(eventArg.id).toEqual(targetId);
```

Comment:
can we use https://jestjs.io/docs/expect#tohavebeennthcalledwithnthcall-arg1-arg2-?

Reply:
sounds good